### PR TITLE
Stripe checkout

### PR DESCRIPTION
**Pull request by AWare**
Changes to support turning on stripe checkout. 
Does not actually turn on stripe checkout.

### CODE REVIEW

File: frontend/assets/javascripts/src/modules/form/paypal.es6 [hunk 1]
```diff
@@ -56,25 +42,11 @@ function createAgreement (paypalData) {
...
-			alert(err);
-		}
-	});
-
+	payment.postForm({ 'payment.payPalBaid': baid.token });
 }
 
 export function init () {
-
+    payment.hello();
```
Comment (rupertbates) [inline frontend/assets/javascripts/src/modules/form/paypal.es6]: Obviously you'll remove this before final PR right ;-)

File: frontend/assets/javascripts/src/modules/payment.es6 [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    form.elems.map(function (elem) {
+        validity.check(elem);
+    });
+
+    let formHasErrors = form.errs.length > 0;
+    return !formHasErrors;
+
+}
+
+// Creates the new member by posting the form data with the BAID.
```
Comment (rupertbates) [inline frontend/assets/javascripts/src/modules/payment.es6]: Change this comment from BAID to payment token, as it now handles Stripe tokens as well right?

File: frontend/assets/javascripts/src/modules/form.js [hunk 3]
```diff
@@ -97,6 +98,11 @@ define([
                 });
 
             }
+            if (form.hasStripeCheckout) {
+                curl('js!https://checkout.stripe.com/checkout.js').then(function () {
```
Comment (AWare) [inline frontend/assets/javascripts/src/modules/form.js]: Done :)

**Comment by Ap0c** [comment]
Looks ok in PROD with test users.

File: frontend/app/views/fragments/form/stripeCheckout.scala.html [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+    guardian.stripeCheckout = {
+        key: guardian.stripePublicKey,
+        image: 'https://d24w1tjgih0o9s.cloudfront.net/gu.png',
+        locale: 'auto',
+        name: 'Guardian',
+        zipCode: false,
+        allowRememberMe: false
+    };
+</script>
+<button type="button" class="action choose-payment-method js-stripe-checkout" data-email="@idUser.email">stripe checkout, yeah</button>
```
Comment (rtyley) [inline frontend/app/views/fragments/form/stripeCheckout.scala.html]: Hey @AWare, can you explain what the `data-email` field is used for? What breaks if I take it away?

File: frontend/assets/javascripts/src/modules/form/stripe.es6 [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    const email = button.data('email');
+    bean.on(window, 'popstate', handler.close);
+
+    const open = (e) => {
+        if (payment.validateForm()) {
+            payment.open();
+            handler.open({
+                description: 'So I could do with some good copy for this, and what it needs to contain like price etc.',
+                panelLabel: 'Approved button copy!',
+                email: email,
```
Comment (rtyley) [inline frontend/assets/javascripts/src/modules/form/stripe.es6]: Oh, so to follow on from https://github.com/guardian/membership-frontend/pull/1436/files#r102215893, after discussion with @Aware it becomes clear that if we don't send an `email` to Stripe, Stripe will require the user to enter their email on the overlay (something we'd obviously rather avoid).

**Pull request merged**